pkg/csi/cinder/controllerserver.go

```diff
@@ -318,7 +318,35 @@ func (cs *controllerServer) GetCapacity(ctx context.Context, req *csi.GetCapacit
 }
 
 func (cs *controllerServer) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("ControllerExpandVolume is not yet implemented"))
+	klog.V(4).Infof("ControllerExpandVolume: called with args %+v", *req)
+
+	volumeID := req.GetVolumeId()
+	if len(volumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
+	}
+	cap := req.GetCapacityRange()
+	if cap == nil {
+		return nil, status.Error(codes.InvalidArgument, "Capacity range not provided")
+	}
+	volSizeBytes := int64(req.GetCapacityRange().GetRequiredBytes())
+	volSizeGB := int(util.RoundUpSize(volSizeBytes, 1024*1024*1024))
+	maxVolSize := cap.GetLimitBytes()
+
+	if maxVolSize > 0 && maxVolSize < volSizeBytes {
+		return nil, status.Error(codes.OutOfRange, "After round-up, volume size exceeds the limit specified")
+	}
```

Comment:
I think we should also respond `OutofRange` in case the required `volSizeBytes ` is larger than cinder maximum limit per volume

Reply:
@adisky I suppose cinder maximum limit per volume is backend dependent , depends on the deployment , may be configured by the user or not. So, ideally cinder handle the errors as per configuration set in cinder? do you think handling it in CSI interface part would be necessary ?

Reply:
@ramineni ya I got it, we can revisit this later, it is not necessary as of now!!